test/unit/app/lib/create-reference.test.js

```diff
@@ -0,0 +1,8 @@
+const createReference = require('../../../../app/lib/create-reference')
+
+describe('Test create reference number', () => {
+  test('Returns reference number', async () => {
```

Comment:
This could be extended by mocking `uuid` import and checking it is using `v4` function to generate the reference.